server/src/main/java/org/opensearch/action/fieldcaps/TransportFieldCapabilitiesIndexAction.java

```diff
@@ -277,7 +278,7 @@ private ShardRouting nextRoutingOrNull(Exception failure) {
         }
 
         private void moveToNextShard() {
-            ++shardIndex;
+            shardIndex.incrementAndGet();
```

Comment:
I don't think the change is needed here, as this won't ever be accessed concurrently.  It's only ever accessed on initialization and then after receiving a response to a transport message, which will incur a network delay.